quinn-proto/src/connection/streams.rs

```diff
@@ -194,10 +211,12 @@ impl Streams {
             }
             Ok(None) => {
                 entry.remove_entry();
+                self.stream_freed(id, StreamHalf::Recv);
```

Comment:
would be nice to somehow combine this with the entry, so that it's harder to miss some places which remove one thing but not the other.

Reply:
I agree in principle, but the obvious approach would involve a newtyped wrapper around the `hash_map::Entry` that captures `self` and therefore makes borrowck sad. We could address that by encapsulating the hash tables along with the set of things `stream_freed` might transitively touch, but it's not clear to me if there's a natural boundary to be drawn there.